src/zenml/integrations/azure/orchestrators/azureml_orchestrator.py

```diff
@@ -0,0 +1,282 @@
+        """
+
+        def _validate_remote_components(
+            stack: "Stack",
+        ) -> Tuple[bool, str]:
+            for component in stack.components.values():
+                if not component.config.is_local:
+                    continue
+
+                return False, (
+                    f"The AzureML orchestrator runs pipelines remotely, "
+                    f"but the '{component.name}' {component.type.value} is "
+                    "a local stack component and will not be available in "
+                    "the AzureML step.\nPlease ensure that you always "
+                    "use non-local stack components with the AzureML "
+                    "orchestrator."
+                )
+
+            return True, ""
+
+        return StackValidator(
+            required_components={
+                StackComponentType.CONTAINER_REGISTRY,
+                StackComponentType.IMAGE_BUILDER,
+            },
+            custom_validation_function=_validate_remote_components,
+        )
+
+    def get_orchestrator_run_id(self) -> str:
+        """Returns the run id of the active orchestrator run.
+
+        Important: This needs to be a unique ID and return the same value for
+        all steps of a pipeline run.
+
+        Returns:
+            The orchestrator run id.
+
+        Raises:
+            RuntimeError: If the run id cannot be read from the environment.
+        """
+        try:
+            return os.environ[ENV_ZENML_AZUREML_RUN_ID]
+        except KeyError:
+            raise RuntimeError(
+                "Unable to read run id from environment variable "
+                f"{ENV_ZENML_AZUREML_RUN_ID}."
+            )
+
+    @staticmethod
+    def _create_command_component(
+        step: Step,
+        step_name: str,
+        image: str,
+        command: List[str],
+        arguments: List[str],
+    ) -> CommandComponent:
+        """Creates a CommandComponent to run on AzureML Pipelines.
+
+        Args:
+            step: The step definition in ZenML.
+            step_name: The name of the step.
+            image: The image to use in the environment
+            command: The command to execute the entrypoint with.
+            arguments: The arguments to pass into the command.
+
+        Returns:
+            the generated AzureML CommandComponent.
+        """
+        env = Environment(image=image)
+
+        outputs = {}
+        if step.config.outputs:
+            outputs = {"completed": Output(type="uri_file")}
+
+        inputs = {}
+        if step.spec.upstream_steps:
+            inputs = {
+                f"{upstream_step}": Input(type="uri_file")
+                for upstream_step in step.spec.upstream_steps
+            }
+
+        return CommandComponent(
+            name=step_name,
+            display_name=step_name,
+            description=f"AzureML CommandComponent for {step_name}.",
+            inputs=inputs,
+            outputs=outputs,
+            environment=env,
```

Comment:
Is this unused by the component, or why do we need to pass the env separately?

Reply:
I am not sure if I get the question correctly here. If you are asking about the env parameter here, that's an AzureML environment. In their own words, environments are:

> an encapsulation of the environment where your machine learning training happens.

We use it to choose the image that the step is running on.

Reply:
Ah my bad, i thought these were the environment variables 😄